### DISCUSSION TITLE
Client & mitmproxy cannot agree on a TLS version to use no matter how tls_version_client_min is updated

### DISCUSSION THREAD

**Discussion by Androo1213** [mitmproxy/mitmproxy#7082] [Intercept Everything!]
I'm trying to read traffic going between my iphone and tandem diabetes pump to hopefully expand the technologies tandem pump users can use in order to manage their diabetes. This is my first time using mitmproxy and upon trying to read traffic from my iphone on my macbook I keep getting warnings suggesting that "Client TLS handshake failed. Client and MITMproxy cannot agree on a TLS version to use. You may need to adjust mitmproxy's tls_version_client_min option."

Although I'm quite new to mitm proxy I've verified that I'm setting up my devices in the exact same ways as documented on mitmproxy's site and that I've enabled full trust for the mitmproxy CA certificate, etc. A developer more familiar with mitmproxy than myself mentioned that there are currently bugs within the documentation online for setting up an ios proxy.

No matter how I vary versions in mitmproxy --set tls_version_client_min=TLS1_0 --set tls_version_client_max=TLS1_3, I can't get it set up so that the client and proxy agree on a TLS version. 

Cheers.

**Comment by tyrylu** [comment]
Same there with iOS 17.6.1, mitmproxy 10.4 with basically any TLS attempts (Weather, random browser), Mitmproxy version info:
Mitmproxy: 10.4.2
Python:    3.12.5
OpenSSL:   OpenSSL 3.3.1 4 Jun 2024
Platform:  Linux-6.10.6-arch1-1-x86_64-with-glibc2.40
Even verbose output does not show anything useful, and I am not sure whether a packet capture of the exchange would help? Or any other debugging output?

**Reply by Androo1213** [reply]
I ended up just using Proxyman and it worked just fine. I think there's some bug with this software right now.

**Comment by tyrylu** [comment]
Happy you could use that, but having a dev comment would be nice anyways.

**Reply by mhils** [reply]
This is hard to comment on - which proxy mode are you using, does plaintext HTTP work, how did you install the mitmproxy CA?

**Comment by tyrylu** [comment]
For the proxy mode, I used defaults, except for a proxy port change. Yes, HTTP works, and the CA install was according to the steps in the doc. However, this will likely not be a CA problem, as the proxy does not get to the point where it would send a cert.

**Reply by mhils** [reply]
Can you 1) double-check that you have enabled "full trust for root certificates" on your iOS device as per https://docs.mitmproxy.org/stable/concepts-certificates/, and 2) if that doesn't fix it, record a PCAP with a failing connection?

**Comment by tyrylu** [comment]
Of course, I checked that, here's one of the attempts: https://oc.trycht.cz/index.php/s/m5w2my7H2QKRPiK Note that the OpenSSL library is a system-wide from Arch Linux repos.

**Reply by mhils** [reply]
Thanks! The PCAP shows TLS 1.0, which will often not be supported anymore by whatever OpenSSL is available on your platform. Unfortunately OpenSSL provides no way for us to determine what's supported by the current build, so tls_version_client_min fails do to what it's supposed to do.

**Reply by mhils** [reply]
https://github.com/mitmproxy/mitmproxy/issues/7138
